loader_hub/remote_depth/base.py

```diff
@@ -0,0 +1,90 @@
+"""Remote file reader.
+
+A loader that fetches an arbitrary remote page or file by URL and parses its contents.
+
+"""
+from typing import Any, Dict, List, Optional, Union
+from urllib.parse import urljoin, urlparse, urlunparse
+
+import requests
+from bs4 import BeautifulSoup
+from llama_index import download_loader
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+from tqdm.auto import tqdm
```

Comment:
Can you lazily import these packages from within the class? For now, it makes it easier to test and also download many loaders at once. Additionally, please create a `requirements.txt` in this directory and drop in bs4 and tqdm.

Reply:
I have lazily imported the packages `bs4, `tqdm` and `urllib.parse`